src/aiohabit/outputs/ansible_inventory.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2020 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Ansible inventory output module."""
+
+from copy import deepcopy
+from aiohabit.outputs.utils import is_windows_host, resolve_hostname
+from aiohabit.utils import get_config_value, get_host_from_metadata, save_yaml
+
+DEFAULT_INVENTORY_PATH = "aiohabit-inventory.yaml"
+DEFAULT_INVENTORY_LAYOUT = {"all": {"children": [], "hosts": {}}}
+
+
+def copy_meta_attrs(host, meta_host, attrs):
+    """Copy attributes from host metadata entry and prefixing them with meta."""
+    for attr in attrs:
+        val = meta_host.get(attr)
+        if val:
+            host[f"meta_{attr}"] = val
+
+
+def find_group(inventory, group_names, top=True):
+    """Find first matching group in inventory which has one of the provided names."""
```

Comment:
If none of the names in `group_names` is found at the inventory and `top` is `True` then all groups will be returned, did I get it right?

Reply:
I'd say that "all" group will be returned. It works with group layout defined in, e.g.: 
```
DEFAULT_INVENTORY_LAYOUT = {"all": {"children": [], "hosts": {}}}
```

But it can be more complex (e.g. for testing FreeIPA) (now defined in provisioning config) where the original code was like:
```
children = {
        'ipa': {
            'children': {
                'ipaserver': {},
                'ipaclient': {},

            }
        },
        'ad': {
            'children': {
                'ad_root': {},
                'ad_subdomain': {},
                'ad_treedomain': {},
                'ad_member': {}
            }
        },
        'linux': {
            'children': {
                'ipa': {},
                'runner': {}
            }
        },
        'windows': {
            'children': {
                'ad': {}
            }
        },

    }
    inventory = {
        'all': {
            'children': children,
            'hosts': {},
        }
    }
```